packages/backend-proxy-middleware/src/base/proxy.ts

```diff
@@ -0,0 +1,351 @@
+ *
+ * @param proxyOptions reference to a proxy options object that the function will enhance
+ * @param backend reference to the backend configuration that the the function may enhance
+ */
+export async function enhanceConfigsForDestination(
+    proxyOptions: Options & { headers: object },
+    backend: DestinationBackendConfig
+): Promise<void> {
+    proxyOptions.target = getDestinationUrlForAppStudio(backend.destination);
+    if (backend.destinationInstance) {
+        proxyOptions.headers[BAS_DEST_INSTANCE_CRED_HEADER] = await getCredentialsForDestinationService(
+            backend.destinationInstance
+        );
+    } else {
+        const destinations = await listDestinations();
+        const destination = destinations[backend.destination];
+        if (destination) {
+            // in case of a full url destination remove the path defined in the destination from the forwarded call
+            if (isFullUrlDestination(destination)) {
+                const destPath = new URL(destination.Host).pathname.replace(/\/$/, '');
+                if (backend.path.startsWith(destPath) && !backend.pathReplace) {
+                    backend.pathReplace = backend.path.replace(destPath, '');
+                }
+            }
+        } else {
+            throw new Error();
+        }
+    }
+}
+
+/**
+ * Enhance the proxy options with information read from the store.
+ *
+ * @param proxyOptions reference to a proxy options object that the function will enhance
+ * @param system backend system information (most likely) read from the store
+ * @param oAuthRequired if true then the OAuth flow is triggered to get cookies
+ * @param tokenChangedCallback function to call if a new refreshToken is available
+ */
+export async function enhanceConfigForSystem(
+    proxyOptions: Options & { headers: object },
+    system: BackendSystem,
+    oAuthRequired: boolean | undefined,
+    tokenChangedCallback: (refreshToken?: string) => void
+): Promise<void> {
+    if (oAuthRequired) {
+        if (system.serviceKeys) {
+            const provider = createForAbapOnBtp(
+                JSON.parse(system.serviceKeys as string),
+                system.refreshToken,
+                tokenChangedCallback
+            );
+            // sending a request to the backend to get cookies
+            await provider.getAtoInfo();
+            proxyOptions.headers['cookie'] = provider.cookies.toString();
+        } else {
+            throw new Error('Cannot connect to ABAP Environment on BTP without service keys.');
+        }
+    } else if (system.authenticationType === AuthenticationType.ReentranceTicket) {
+        throw new Error('Feature comes with https://github.com/SAP/open-ux-tools/pull/485');
+        /*
+        const provider = createForAbapOnBtp(
+            ...
+        );
+        // sending a request to the backend to get cookies
+        await provider.getAtoInfo();
+        proxyOptions.headers['cookie'] = provider.cookies.toString();
+        */
```

Comment:
Can we remove this one?

Reply:
I wanted to keep it for @ullasholla since he needs to add it with #485